README.md
I just fixed it for WoW:BFA.

The Addon was original created by: Kruithne

 

Issues: https://github.com/Lonermax/PremadeAutoAccept


 Original description:

== PremadeAutoAccept ==

This project is open-source and maintained on GitHub here: https://github.com/Kruithne/PremadeAutoAccept

This simple add-on for World of Warcraft adds back in the auto-accept functionality that Blizzard removed from the premade group finder.

CHANGELOG.md
** Version 1.3.3 **
* toc push *

** Version 1.3.2 **
* NOT TESTED!
* Debug for Shadowlands pre-patch (credit: https://github.com/virtualbs) 


** Version 1.3.1 **
* Small UI tweak (itemlevel or private mode coverd the text.)

** Version 1.3 **
* Curse release.

** Version 1.2.3 **
* Fix: Auto invite after 1 player left at full party.
* some debug code.

** Version 1.2.2 **
* Enabled auto-invite for +5 groups again. // spam bug seems fixed.

** Version 1.2.1 **
* Disabled auto-invite for 5+ groups. Till fixed found for RaidConvert spaming.
* sorry for that =(

** Version 1.2 **
* Updated toc.
* BugFixed: double invite people from list.

** Version 1.1 **
* Cleaned up implementation to no longer rely on hacky code.
* Added role-specific auto-inviting, since we're at it.

** Version 1.0 **
* The auto-accept feature for premade group finder was removed, and this add-on was made.

PremadeAutoAcceptFix.lua
--[[
	I just fixed it. -> Lonermax
	https://github.com/Lonermax/PremadeAutoAccept


	Original created by:
	PremadeAutoAccept (C) Kruithne <kruithne@gmail.com>
	Licensed under GNU General Public Licence version 3.

	https://github.com/Kruithne/PremadeAutoAccept
]]--

local eventFrame = CreateFrame("FRAME");
local isAutoAccepting = false;
local displayedRaidConvert = false;
local autoAccepting = {};

local function InviteApplicants()
	local applicants = C_LFGList.GetApplicants();
	for i = 1, #applicants do
		local applicantData = C_LFGList.GetApplicantInfo(applicants[i]);

		-- Using the premade "invite" feature does not work, as Blizzard have broken auto-accept intentionally
		-- Because of this, we can't invite groups, but we can still send normal invites to singletons.
		if applicantData and (applicantData.applicationStatus == "applied" or applicantData.pendingApplicationStatus == "applied") and applicantData.numMembers == 1 then

			local name, _, _, _, _, _, _, _, _, assignedRole  = C_LFGList.GetApplicantMemberInfo(applicants[i], 1);
			if autoAccepting[assignedRole] then
				C_PartyInfo.InviteUnit(name);
			end
		end
	end
end

local function OnCheckBoxClick(self)
	isAutoAccepting = self:GetChecked();
	autoAccepting[self.role] = isAutoAccepting;

	if isAutoAccepting then
		PlaySound(SOUNDKIT.IG_MAINMENU_OPTION_CHECKBOX_ON);
		InviteApplicants();
	else
		PlaySound(SOUNDKIT.IG_MAINMENU_OPTION_CHECKBOX_OFF);
	end
end

local function CreateCheckbox(atlas, role)
	local button = CreateFrame("CheckButton", nil, LFGListFrame.ApplicationViewer);
	button:SetWidth(22);
	button:SetHeight(22);
	button:Show();

	button:SetNormalTexture("Interface\\Buttons\\UI-CheckBox-Up");
	button:SetPushedTexture("Interface\\Buttons\\UI-CheckBox-Down");
	button:SetHighlightTexture("Interface\\Buttons\\UI-CheckBox-Highlight");
	button:SetCheckedTexture("Interface\\Buttons\\UI-CheckBox-Check");

	button.role = role;
	button:SetScript("OnClick", OnCheckBoxClick);

	local icon = button:CreateTexture(nil, "ARTWORK");
	icon:SetAtlas(atlas);
	icon:SetWidth(17);
	icon:SetHeight(17);
	icon:SetPoint("LEFT", button, "RIGHT", 2, 0);
	button.icon = icon;

	return button;
end

local function CreateAutoAcceptButtons()
	local header = LFGListFrame.ApplicationViewer:CreateFontString(nil, "ARTWORK", "GameFontHighlightSmall");
	header:SetPoint("BOTTOMLEFT", LFGListFrame.ApplicationViewer.InfoBackground, "BOTTOMLEFT", 12, 25);
	header:SetText(LFG_LIST_AUTO_ACCEPT);
	header:SetJustifyH("LEFT");

	local damageButton = CreateCheckbox("groupfinder-icon-role-large-dps", "DAMAGER");
	damageButton:SetPoint("TOPLEFT", header, "BOTTOMLEFT", 0, -1);

	local healerButton = CreateCheckbox("groupfinder-icon-role-large-heal", "HEALER");
	healerButton:SetPoint("LEFT", damageButton.icon, "RIGHT", 5, 0);

	local tankButton = CreateCheckbox("groupfinder-icon-role-large-tank", "TANK");
	tankButton:SetPoint("LEFT", healerButton.icon, "RIGHT", 5, 0);
end

local function OnLoad()
	eventFrame:RegisterEvent("LFG_LIST_APPLICANT_LIST_UPDATED");
	eventFrame:RegisterEvent("GROUP_LEFT");
	eventFrame:RegisterEvent("PARTY_LEADER_CHANGED");

	CreateAutoAcceptButtons();
end

local function OnApplicantListUpdated()
	if UnitIsGroupLeader("player", LE_PARTY_CATEGORY_HOME) then
		if isAutoAccepting then
			-- Display conversion to raid notice.
			if not displayedRaidConvert and not IsInRaid(LE_PARTY_CATEGORY_HOME) then
				local futureCount = GetNumGroupMembers(LE_PARTY_CATEGORY_HOME) + C_LFGList.GetNumInvitedApplicantMembers() + C_LFGList.GetNumPendingApplicantMembers();
				if futureCount > (MAX_PARTY_MEMBERS + 1) then
					StaticPopup_Show("LFG_LIST_AUTO_ACCEPT_CONVERT_TO_RAID");
					displayedRaidConvert = true;
				end
			end
			-- tried to fix the raid convert spam bug
			if displayedRaidConvert and not IsInRaid(LE_PARTY_CATEGORY_HOME) then
				local futureCount = GetNumGroupMembers(LE_PARTY_CATEGORY_HOME) + C_LFGList.GetNumInvitedApplicantMembers() + C_LFGList.GetNumPendingApplicantMembers();
				if futureCount < 6 then
					InviteApplicants();
					do return end;
				else
					-- print("DEBUG:"..futureCount);
					do return end;
				end 
			end
			InviteApplicants();
		end
	end
end

local function OnEvent(self, event, ...)
	if event == "ADDON_LOADED" then
		local addonName = ...;
		local ADDON_NAME = "PremadeAutoAcceptFix";

		if addonName == "PremadeAutoAcceptFix" then
			OnLoad();
			self:UnregisterEvent("ADDON_LOADED");
		end
	elseif event == "LFG_LIST_APPLICANT_LIST_UPDATED" or event == "PARTY_LEADER_CHANGED" or event == "GROUP_ROSTER_UPDATE" then
		OnApplicantListUpdated();
	elseif event == "GROUP_LEFT" then
		isAutoAccepting = false;
		displayedRaidConvert = false;
	end
end

eventFrame:RegisterEvent("ADDON_LOADED");
eventFrame:SetScript("OnEvent", OnEvent);


PremadeAutoAccept.lua
--[[
	PremadeAutoAccept (C) Kruithne <kruithne@gmail.com>
	Licensed under GNU General Public Licence version 3.

	https://github.com/Kruithne/PremadeAutoAccept
]]--

local eventFrame = CreateFrame("FRAME");
local isAutoAccepting = false;
local displayedRaidConvert = false;
local autoAccepting = {};

local function InviteApplicants()
	local applicants = C_LFGList.GetApplicants();
	for i = 1, #applicants do
		local applicantData = C_LFGList.GetApplicantInfo(applicants[i]);

		-- Using the premade "invite" feature does not work, as Blizzard have broken auto-accept intentionally
		-- Because of this, we can't invite groups, but we can still send normal invites to singletons.
		if applicantData and (applicantData.applicationStatus == "applied" or applicantData.pendingApplicationStatus == "applied") and applicantData.numMembers == 1 then

			local name, _, _, _, _, _, _, _, _, assignedRole  = C_LFGList.GetApplicantMemberInfo(applicants[i], 1);
			if autoAccepting[assignedRole] then
				InviteUnit(name);
			end
		end
	end
end

local function OnCheckBoxClick(self)
	isAutoAccepting = self:GetChecked();
	autoAccepting[self.role] = isAutoAccepting;

	if isAutoAccepting then
		PlaySound(SOUNDKIT.IG_MAINMENU_OPTION_CHECKBOX_ON);
		InviteApplicants();
	else
		PlaySound(SOUNDKIT.IG_MAINMENU_OPTION_CHECKBOX_OFF);
	end
end

local function CreateCheckbox(atlas, role)
	local button = CreateFrame("CheckButton", nil, LFGListFrame.ApplicationViewer);
	button:SetWidth(22);
	button:SetHeight(22);
	button:Show();

	button:SetNormalTexture("Interface\\Buttons\\UI-CheckBox-Up");
	button:SetPushedTexture("Interface\\Buttons\\UI-CheckBox-Down");
	button:SetHighlightTexture("Interface\\Buttons\\UI-CheckBox-Highlight");
	button:SetCheckedTexture("Interface\\Buttons\\UI-CheckBox-Check");

	button.role = role;
	button:SetScript("OnClick", OnCheckBoxClick);

	local icon = button:CreateTexture(nil, "ARTWORK");
	icon:SetAtlas(atlas);
	icon:SetWidth(17);
	icon:SetHeight(17);
	icon:SetPoint("LEFT", button, "RIGHT", 2, 0);
	button.icon = icon;

	return button;
end

local function CreateAutoAcceptButtons()
	local header = LFGListFrame.ApplicationViewer:CreateFontString(nil, "ARTWORK", "GameFontHighlightSmall");
	header:SetPoint("BOTTOMLEFT", LFGListFrame.ApplicationViewer.InfoBackground, "BOTTOMLEFT", 12, 30);
	header:SetText(LFG_LIST_AUTO_ACCEPT);
	header:SetJustifyH("LEFT");

	local damageButton = CreateCheckbox("groupfinder-icon-role-large-dps", "DAMAGER");
	damageButton:SetPoint("TOPLEFT", header, "BOTTOMLEFT", 0, -1);

	local healerButton = CreateCheckbox("groupfinder-icon-role-large-heal", "HEALER");
	healerButton:SetPoint("LEFT", damageButton.icon, "RIGHT", 5, 0);

	local tankButton = CreateCheckbox("groupfinder-icon-role-large-tank", "TANK");
	tankButton:SetPoint("LEFT", healerButton.icon, "RIGHT", 5, 0);
end

local function OnLoad()
	eventFrame:RegisterEvent("LFG_LIST_APPLICANT_LIST_UPDATED");
	eventFrame:RegisterEvent("GROUP_LEFT");
	eventFrame:RegisterEvent("PARTY_LEADER_CHANGED");

	CreateAutoAcceptButtons();
end

local function OnApplicantListUpdated()
	if UnitIsGroupLeader("player", LE_PARTY_CATEGORY_HOME) then
		if isAutoAccepting then
			-- Display conversion to raid notice.
			if not displayedRaidConvert and not IsInRaid(LE_PARTY_CATEGORY_HOME) then
				local futureCount = GetNumGroupMembers(LE_PARTY_CATEGORY_HOME) + C_LFGList.GetNumInvitedApplicantMembers() + C_LFGList.GetNumPendingApplicantMembers();
				if futureCount > (MAX_PARTY_MEMBERS + 1) then
					StaticPopup_Show("LFG_LIST_AUTO_ACCEPT_CONVERT_TO_RAID");
					displayedRaidConvert = true;
				end
			end
			-- tried to fix the raid convert spam bug
			if displayedRaidConvert and not IsInRaid(LE_PARTY_CATEGORY_HOME) then
				local futureCount = GetNumGroupMembers(LE_PARTY_CATEGORY_HOME) + C_LFGList.GetNumInvitedApplicantMembers() + C_LFGList.GetNumPendingApplicantMembers();
				if futureCount < 6 then
					InviteApplicants();
					do return end;
				else
					-- print("DEBUG:"..futureCount);
					do return end;
				end 
			end
			InviteApplicants();
		end
	end
end

local function OnEvent(self, event, ...)
	if event == "ADDON_LOADED" then
		local addonName = ...;
		local ADDON_NAME = "PremadeAutoAccept";

		if addonName == "PremadeAutoAccept" then
			OnLoad();
			self:UnregisterEvent("ADDON_LOADED");
		end
	elseif event == "LFG_LIST_APPLICANT_LIST_UPDATED" or event == "PARTY_LEADER_CHANGED" or event == "GROUP_ROSTER_UPDATE" then
		OnApplicantListUpdated();
	elseif event == "GROUP_LEFT" then
		isAutoAccepting = false;
		displayedRaidConvert = false;
	end
end

eventFrame:RegisterEvent("ADDON_LOADED");
eventFrame:SetScript("OnEvent", OnEvent);
